components/inputs/input-radio.js

```diff
@@ -0,0 +1,177 @@
+import { css, html, LitElement, nothing } from 'lit';
+import { classMap } from 'lit/directives/class-map.js';
+import { FocusMixin } from '../../mixins/focus/focus-mixin.js';
+import { getUniqueId } from '../../helpers/uniqueId.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { InputInlineHelpMixin } from './input-inline-help.js';
+import { PropertyRequiredMixin } from '../../mixins/property-required/property-required-mixin.js';
+import { radioStyles } from './input-radio-styles.js';
+import { SkeletonMixin } from '../skeleton/skeleton-mixin.js';
+
+/**
+ * A radio input within a <d2l-input-radio-group>.
+ * @slot inline-help - Help text that will appear below the input. Use this only when other helpful cues are not sufficient, such as a carefully-worded label.
+ * @slot supporting - Supporting information which will appear below and be aligned with the input.
+ */
+class InputRadio extends InputInlineHelpMixin(SkeletonMixin(FocusMixin(PropertyRequiredMixin(LitElement)))) {
+
+	static get properties() {
+		return {
+			/**
+			 * Checked state
+			 * @type {boolean}
+			 */
+			checked: { type: Boolean, reflect: true },
+			/**
+			 * ACCESSIBILITY: Additional information communicated to screenreader users when focusing on the input
+			 * @type {string}
+			 */
+			description: { type: String },
+			/**
+			 * Disables the input
+			 * @type {boolean}
+			 */
+			disabled: { type: Boolean, reflect: true },
+			/**
+			 * REQUIRED: Label for the input
+			 * @type {string}
+			 */
+			label: { required: true, type: String },
+			/**
+			 * Hides the supporting slot when unchecked
+			 * @type {boolean}
+			 */
+			supportingHiddenWhenUnchecked: { type: Boolean, attribute: 'supporting-hidden-when-unchecked', reflect: true },
```

Comment:
This is to make showing supporting content only when checked easier -- I saw a few consumers doing this manually.

![supporting](https://github.com/user-attachments/assets/07b595cb-592d-4708-8d6a-c14346d21ffc)

Reply:
Nice!